Makefile

```diff
@@ -1,8 +1,59 @@
-test_mock:
-	go test -tags ledger_mock
+include common.mk
 
-test_zemu:
-	go test -tags ledger_zemu
+# Set all target as the default target.
+all: build
 
-test_hid:
-	go test
+# Build.
+build:
+	@$(ECHO) "$(MAGENTA)*** Building Go code...$(OFF)"
+	@$(GO) build -v ./...
+
+# Format code.
+fmt:
+	@$(ECHO) "$(CYAN)*** Running Go formatters...$(OFF)"
+	@gofumpt -s -w .
+	@gofumports -w -local github.com/oasisprotocol/oasis-core-ledger .
+
+# Lint code, commits and documentation.
+lint-targets := lint-go lint-docs lint-git lint-go-mod-tidy
+
+lint-go:
+	@$(ECHO) "$(CYAN)*** Running Go linters...$(OFF)"
+	@env -u GOPATH golangci-lint run
+
+lint-git:
+	@$(ECHO) "$(CYAN)*** Runnint gitlint...$(OFF)"
+	@$(CHECK_GITLINT)
+
+lint-docs:
+	@$(ECHO) "$(CYAN)*** Runnint markdownlint-cli...$(OFF)"
+	@npx markdownlint-cli '**/*.md'
+
+lint-go-mod-tidy:
+	@$(ECHO) "$(CYAN)*** Checking go mod tidy...$(OFF)"
+	@$(ENSURE_GIT_CLEAN)
+	@$(CHECK_GO_MOD_TIDY)
+
+lint: $(lint-targets)
+
+# Test.
+test-targets := test-unit
+
+test-unit:
```

Comment:
What about the other test targets (e.g., we now have the `ledgerhw` tag for tests requiring actual hardware), the previous makefile also had other test targets but not sure if those actually did anything?

Reply:
Yes, makes sense to try that. I'll add that in a separate PR.